layers/+lang/c-c++/packages.el

```diff
@@ -279,33 +280,30 @@
       (spacemacs/set-leader-keys-for-major-mode mode "gi" 'cscope-index-files))))
 
 ;; BEGIN LSP BACKEND PACKAGES
+(defun c-c++/post-init-lsp-mode ()
+  (when (eq c-c++-backend 'lsp)
+    (spacemacs//c-c++-lsp-config)
+    (unless (eq c-c++-lsp-server 'clangd)
+      (require c-c++-lsp-server) ; i.e. the 'ccls or 'cquery package
+      (remhash 'clangd lsp-clients))
+    (spacemacs/add-to-hooks 'lsp c-c++-mode-hooks)))
```

Comment:
In the spirit of spacemacs of lazy loading everything, I suggest using `lsp-deferred` here instead of `lsp`, this will prevent clangd from starting if there's no C++ buffer on any window. @yyoncho What do you think?

Reply:
I think that it should be consistent across all lsp enabled layers, so IMO it should be pushed up in lsp layer and it should be configurable - thus it is out of scope for this PR.